### DISCUSSION TITLE
Failed to get -2 sigma of sensitivity band.

### DISCUSSION THREAD

**Discussion by Feiyuemoxuan** [histfitter/histfitter#199] [General]
Hello everyone. 
I tried the asymptotic and frequentist calculators, and both methods could not give the -2 sigma limit of the sensitivity band (or it may mean the -2 sigma refers to a negative mu?). Here, I show the result of the frequentist calculator. I tried different signal models, and such a phenomenon is always shown in the final results. Are there any necessary configurations of HistFitter?

![pvalue_vs_mu](https://github.com/histfitter/histfitter/assets/29231228/b96edf0d-43a7-4339-ac3d-4070e7e2a6fe)

**Comment by longjon929** [comment]
Hello @Feiyuemoxuan ,

-2 sigma is where the yellow band crosses 0.05 on the left side (at least for 95% CL), ie below the median expected value.  Why is the red line drawn here at 0.1?  Are you doing 90% CL limits?

Could you also include the asymptotic plot?  Since the issue is at mu=0, different signal models shouldn't make a difference.  How many toys did you use?  You could try with a bit more, as the band is a little choppy, but it probably won't resolve this since you see it in asymptotics.  Does your model have large systematic errors?  It looks to me like your model has large uncertainties or low stats and can't constrain the interval at 2 sigma above 0.  I haven't thought about how to interpret that.

**Reply by Feiyuemoxuan** [reply]
Hello @longjon929 , thanks for your reply!

1. I am doing 90% CL limits. I am working on a dark matter detection experiment, different from collider experiments, and we use 90% CL limits;
2. The asymptotic plot:
<img width="930" alt="image" src="https://github.com/histfitter/histfitter/assets/29231228/6128c798-32db-4fa7-b95a-95b6b03dd87f">
3. 1000 toys are used;
4. You are right ! indeed my model has large systematic errors. One of the fit components is set to "free" (the nuisance is set to [0, 2]). I am trying to play with different uncertainties. Thanks very much!

**Reply by Feiyuemoxuan** [reply]
Is there any other way to set a nuissance to be absolutely free? Here shows my current code:
`nuissance = Systematic("nuissance", configMgr.weights, 1 + 1.0, 1 - 1.0, "user", "userOverallSys")`

**Comment by longjon929** [comment]
Hi @Feiyuemoxuan ,

Ah ok, that makes sense.  If you don't mind, I'm curious which experiment?

It would be easier to tell with the yields used in your fit, but I believe your upper limit plot is telling you that if the observed background were to be more then 1 sigma below expected (I believe that's the right direction...), you would expect to set a very small upper limit.

Adding a systematic like that is still going to result in a nuisance parameter that has a Gaussian constraint, albeit with a 100% uncertainty.  For an unconstrainted parameter, you'll want to use [setNormFactor](https://github.com/histfitter/histfitter/blob/master/python/sample.py#L1238), similar to how you added 'mu' for your signal.  Do you have a region to perform some sort of auxiliary measurement to determine this parameter's value?  I'm not sure how your fit is setup, but if you have more degrees of freedom in the fit than regions to constrain it, the fit will likely have problems converging.

Best,
Jonathan

**Reply by Feiyuemoxuan** [reply]
Hi @longjon929, thanks again.I am working in PandaX. 

I used setNormFactor and it has very small impact on final results;
I don't have auxiliary measurement to determine the parameter. I tried to set very small uncertainties for other nuisances, but the sensitivity band is still very broad。Also, I try to add small uncertainty for the free parameter mentioned before and the results are same.

**Reply by longjon929** [reply]
Hi @Feiyuemoxuan,

What is your expected yield like?  If it's small you'll have large relative statistical uncertainties.  You could try with only statistical uncertainties and no systematics to see what it looks like.

**Reply by Feiyuemoxuan** [reply]
Sorry to reply so late, @longjon929, I just finished my vacation. 
Do you mean the expected yield of the signal? The yield of the signal is small indeed. For backgrounds, some are big and some are small. My systematic uncertainties are small, so turning off the systematics does not influence the results.

**Reply by longjon929** [reply]
Hi @Feiyuemoxuan ,

I mean the expected background yield.  Do you have numbers for expected signal yield at mu=1 (and what is the uncertainty on that number) and your expected background yield +/- total uncertainty?  Those would be helpful to understand your limit plot.  

How many free parameters do you have in your fit, and how many constraints?  It could be that your model is underconstrained.

**Reply by Feiyuemoxuan** [reply]
The signal is set to free in my fitting. The input (mu=1) signal yield is about 2 events and the background is about 1190 +/- 52 events. 5 free parameters and 14 constrained parameters.

**Reply by longjon929** [reply]
Hi @Feiyuemoxuan ,

Edit: I forgot to update the systematic uncertainties in the test file, so I've updated the numbers below.

I can't say I understand how you're getting an upper limit around mu=2, when the uncertainty on your background is much larger than your signal.  If I just plug the numbers you gave me into the tutorial MyUserAnalysis.py:

```
# Set observed and expected number of events in counting experiment
ndata     =  1190. 	# Number of events observed in data
nbkg      =  1190.	 	# Number of predicted bkg events
nsig      =  2.  	# Number of predicted signal events
nbkgErr   =  1.  	# (Absolute) Statistical error on bkg estimate
nsigErr   =  0.1  	# (Absolute) Statistical error on signal estimate
lumiError = 0.039 	# Relative luminosity uncertainty
ucb = Systematic("ucb", configMgr.weights, 1.043,0.957, "user","userOverallSys") # only apply this for 52/1190

```
then I get a limit around 30
![image](https://github.com/histfitter/histfitter/assets/1594459/e60bb221-3b9d-4bfc-b30a-f02244afe722)

Even if you remove the systematic on the background, the statistical error alone on your expected background is going to be large relative to your signal and lead to large upper limits:
![image](https://github.com/histfitter/histfitter/assets/1594459/fa784a7f-d414-4f4d-8bdd-b09858e97592)

So I suspect something weird is going on with the 5 floating parameters.  Have you run a simple fit to see what they minimize to and check that your fit is converging properly?

**Reply by Feiyuemoxuan** [reply]
Thanks for your nice suggestions! Let me try to simplify the fit first. Currently, the log shows my fits have finally converged and reached a valid minimum.
